docs/appendices/release-notes/unreleased.rst

```diff
@@ -76,6 +76,9 @@ Breaking Changes
   :ref:`named index column definition <named-index-column>` instead of
   silently ignoring them.
 
+- Reduced the maximum allowed array index from ``Integer.MAX_VALUE + 1`` to
```

Comment:
If we add this to breaking changes then we cannot backport, I'd say not to mention it as breaking change, but please ask for another opinion on this.

Reply:
Hello @mfussenegger , could you share your thoughts on this? This looks like a breaking change but under the hood it is actually not.

Reply:
Thanks. If I understood correctly, I would split this and consider the max value portion as a change that would not be backported and the remainder as a bug fix, right?

Reply:
I'd treat both as a change. From a user perspective the change never _solves_ a problem, it only helps spotting a potential problem.